app.go

```diff
@@ -72,6 +72,90 @@ type MagicLink struct {
 	URL         string `json:"url"`
 }
 
+type AppInfo struct {
+	Name                       string              `json:"name"`
+	ID                         string              `json:"id"`
+	AuthOrigin                 string              `json:"auth_origin"`
+	RedirectURL                string              `json:"redirect_url"`
+	LoginURL                   string              `json:"login_url"`
+	PublicKey                  string              `json:"rsa_public_key"`
+	AllowedIdentifier          string              `json:"allowed_identifier"`
+	RequiredIdentifier         string              `json:"required_identifier"`
+	RequireEmailVerification   bool                `json:"require_email_verification"`
+	SessionTimeoutLength       int                 `json:"session_timeout_length"`
+	Role                       string              `json:"role,omitempty"`
+	UserMetadataSchemaResponse []UserMetadataField `json:"user_metadata_schema"`
+	Layouts                    Layouts             `json:"layouts"`
+}
+type UserMetadataField struct {
+	Handle       string                `json:"id"`
+	FieldName    string                `json:"field_name"`
+	FieldType    UserMetadataFieldType `json:"type"`
+	FriendlyName string                `json:"friendly_name"`
+	Registration bool                  `json:"registration"`
+	Profile      bool                  `json:"profile"`
+}
+
+type CreateUserMetadataField struct {
+	FriendlyName string                `json:"friendly_name,omitempty"`
+	FieldType    UserMetadataFieldType `json:"type,omitempty"`
+	Registration bool                  `json:"registration,omitempty"`
+	Profile      bool                  `json:"profile,omitempty"`
+}
+
+type UpdateUserMetadataField struct {
+	FriendlyName string `json:"friendly_name,omitempty"`
+	Registration *bool  `json:"registration,omitempty"`
+	Profile      *bool  `json:"profile,omitempty"`
+}
+
+type UserMetadataFieldType string
+
+const (
+	StringMD  UserMetadataFieldType = "string"
+	BooleanMD UserMetadataFieldType = "boolean"
+	NumberMD  UserMetadataFieldType = "integer"
+	DateMD    UserMetadataFieldType = "date"
+	PhoneMD   UserMetadataFieldType = "phone"
+	EmailMD   UserMetadataFieldType = "email"
+)
+
+type Layouts struct {
+	Registration []LayoutConfig `json:"registration"`
+	Profile      []LayoutConfig `json:"profile"`
+}
+
+type LayoutConfig struct {
+	ID string `json:"id"`
+	X  uint   `json:"x"`
+	Y  uint   `json:"y"`
+	W  uint   `json:"w"`
+	H  uint   `json:"h"`
+}
+
+// GetApp gets information about an app
+// returns App on success, error on failure
+func (a *App) GetApp() (*AppInfo, error) {
+	type respAppInfo struct {
+		App AppInfo `json:"app"`
+	}
+	var appResp respAppInfo
+
+	response, err := resty.New().R().
+		SetResult(&appResp).
+		SetAuthToken(a.Config.APIKey).
```

Comment:
Why are we setting an auth token for an unprotected route? We don't need an authorization header to access the endpoint:
`https://api.passage.id/v1/apps/{appID}`

Reply:
good call